src/core/qgsvectorlayer.cpp

```diff
@@ -178,6 +178,11 @@ QgsVectorLayer::QgsVectorLayer( const QString &vectorLayerPath,
     setDataSource( vectorLayerPath, baseName, providerKey, providerOptions, options.loadDefaultStyle );
   }
 
+  for ( const QgsField &field : mFields )
```

Comment:
```suggestion
  for ( const QgsField &field : qgis::as_const( mFields ) )
```

Reply:
Does it make sense to update other occurences?

Reply:
If they are in non const methods, absolutely. If they are in const methods it's not required (but I think it would not hurt and definitely ease reviewing)